python/paddle/v2/framework/tests/test_tensor.py

```diff
@@ -43,6 +43,63 @@ def test_float_tensor(self):
         self.assertAlmostEqual(1.0, tensor_array_2[3, 9])
         self.assertAlmostEqual(2.0, tensor_array_2[19, 11])
 
+    def test_int_lod_tensor(self):
+        scope = core.Scope()
+        var = scope.new_var("test_tensor")
+        var_lod = scope.new_var("test_lod_tensor")
+        place = core.CPUPlace()
+
+        tensor = var.get_tensor()
+        lod_tensor = var_lod.get_lod_tensor()
+
+        tensor.set_dims([4, 4, 6])
+        tensor.alloc_int(place)
+        array = numpy.array(tensor)
+        array[0, 0, 0] = 3
+        array[3, 3, 5] = 10
+        tensor.set(array, place)
+
+        lod_tensor.set_tensor(tensor)
+        lod_tensor.set_lod([[0, 2, 4]])
+
+        lod_v = numpy.array(lod_tensor.tensor())
+        self.assertTrue(numpy.alltrue(array == lod_v))
+
+        lod = lod_tensor.lod()
+        self.assertEqual(0, lod[0][0])
+        self.assertEqual(2, lod[0][1])
+        self.assertEqual(4, lod[0][2])
+
+    def test_float_lod_tensor(self):
+        scope = core.Scope()
+        var = scope.new_var("test_tensor")
+        var_lod = scope.new_var("test_lod_tensor")
```

Comment:
I am wondering do we need two variables to represent a LodTensor in operators, and how tensor memory is managed.

I try to implement infer shape in add two operator:
```
void InferShape(const framework::InferShapeContext &ctx) const override {
    auto tensor_x = ctx.Input<LoDTensor>("X")->tensor();
    auto tensor_y = ctx.Input<LoDTensor>("Y")->tensor();
    PADDLE_ENFORCE_EQ(tensor_x.dims(),
                      tensor_y.dims(),
                      "Two input of Add Op's dimension must be same.");
    auto* lod_tensor_out =   ctx.Output<LodTensor>("Out");
    Tensor* out = new Tensor();
    out->Resize(tensor_x->dims());
    lod_tensor_out->set_tensor(out);
  }
};
```

It seems that it's better to manage the tensor memory in LoDTensor.  Or we have to use shared_ptr in LodTensor.

```
class LodTensor {
  shared_ptr<Tensor> tensor_;
  Lod lod_;
};
```

Reply:
@QiJune Good questions, Maybe `class LodTensor : : public Tensor { } ` is betther to avoid use `std::unique_ptr` or `std::shared_ptr` ?

Reply:
If using `std::unique_ptr`, we can fix it in another PR.  If using `class LodTensor  : public Tensor { }`, I can change in this PR.

Reply:
yes, tensor should be a part of LODTensor.

a raw pointer with a bool flag indicating whether this LoDTensor owns the memory seems better.

If some else what to share this Tensor, just copy the pointer.

Reply:
@Superjom @QiJune @reyoung This problem will be fixed in next PR.